docs/designs/transaction-design.md

```diff
@@ -0,0 +1,222 @@
+The Transaction History system consists of:  
+- A SQL Server table for storing transaction records.
+- API endpoints for CRUD transaction data.
+- A front-end page for displaying transaction history to admin users.
+
+## Architecture Goals and Constraints
+
+### Goals
+1. Ensure transaction history is recorded with:
+    - Primary key for each transaction.
+    - User ID.
+    - Transaction GUID.
+    - Item ID.
+    - Transaction type (e.g., Add, Checkout).
+    - Total amount.
+    - Transaction date.
+2. Provide API endpoints for recording and retrieving transaction data only for admin users.
+
+### Constraints
+- Admin role management will control access to the transaction history data.
+- Transaction data must be logged in real-time during any item addition or checkout.
+- The transaction history page should be able to display all transactions for a specific user.
+- The transaction history page should be able to display detailed information for each transaction.
+- The transaction history page should be able to filter and search transactions based on various criteria.
+
+## Detailed Design  
+
+### Database Schema
+
+#### Database  
+
+**Tables:**  
+- **TransactionHistory**:  
+  - `TransactionID` (Primary Key) - Unique identifier for each transaction record (UUID or GUID).
+  - `UserID` (Foreign Key) - References the `Users` table to identify the admin user associated with the transaction.
+  - `TransactionGUID`- Unique identifier for each transaction session (UUID or GUID). Used to correlate multiple entries for the same transaction (e.g., for the same user interacting multiple times within one session).
+  - `ItemID` (Foreign Key): References the `Items` table to identify the item associated with the transaction.
+  - `TransactionType`: VARCHAR(50), Type of transaction (e.g., Add, Checkout).
+  - `TotalAmount`: DECIMAL(10, 2), Total amount of the transaction.
+  - `TransactionDate`: DATETIME, Date and time when the transaction occurred.
+
+**SQL Schema:**
+```sql
+CREATE TABLE TransactionHistory (
+    TransactionID UNIQUEIDENTIFIER PRIMARY KEY,       
+    UserID INT,                                        
+    TransactionGUID UNIQUEIDENTIFIER,                  
+    ItemID INT,                                         
+    TransactionType VARCHAR(50),                          
+    TotalAmount DECIMAL(10, 2),                         
+    TransactionDate DATETIME DEFAULT GETDATE(),         
+    
+    -- Define foreign key constraints
+    FOREIGN KEY (UserID) REFERENCES Users(UserID),      
+    FOREIGN KEY (ItemID) REFERENCES Items(ItemID)       
+);
+
+-- Optionally, we may want to create an index for faster searching by TransactionGUID, UserID, and TransactionDate
+CREATE INDEX IX_TransactionHistory_TransactionGUID ON TransactionHistory(TransactionGUID);
+CREATE INDEX IX_TransactionHistory_UserID ON TransactionHistory(UserID);
+CREATE INDEX IX_TransactionHistory_TransactionDate ON TransactionHistory(TransactionDate);
```

Comment:
Great idea. Let's wait till we hit performance issues though. But this is the one table that will get very long quickly.